### DISCUSSION TITLE
How to add internal command used to enter system shell

### DISCUSSION THREAD

**Discussion by marklion** [daniele77/cli#213] [Q&A] [answered]
I am trying to add an internal command like 'bash', which should be used to open an system shell like '/bin/bash' from current terminal. But it seems not working as my expect when I write this kind of code below.
```
root_menu->Insert(
        "bash", [](std::ostream &out)
        { system("/bin/bash"); });

The process exit after I issued 'bash' command.
```

Thanks for your kindly help.

**Comment by daniele77** [comment, marked as answer]
Are you really sure that the process exited?
When you call `system("/bin/bash")`, you're getting a brand new bash prompt, which might seem like the process exited.
Write "exit" to exit the new bash shell you just created using the bash command. It should return to the cli application.
